main/src/com/google/refine/expr/functions/Cross.java

```diff
@@ -41,18 +41,25 @@
 import com.google.refine.expr.WrappedCell;
 import com.google.refine.grel.ControlFunctionRegistry;
 import com.google.refine.grel.Function;
+import com.google.refine.model.Project;
 import com.google.refine.util.GetProjectIDException;
 import com.google.refine.util.LookupException;
 
 public class Cross implements Function {
 
+    public static final String INDEX_COLUMN_NAME = "_OpenRefine_Index_Column_Name_";
+
     @Override
     public Object call(Properties bindings, Object[] args) {
-        if (args.length == 3) {
+        if (args.length == 1 || args.length == 2 || args.length == 3) {
             // 1st argument can take either value or cell(for backward compatibility)
             Object v = args[0];
-            Object targetProjectName = args[1];
-            Object targetColumnName = args[2];
+            // if 2nd argument is omitted or set to "", use the current project name
+            Object targetProjectName = args.length < 2 || args[1].equals("") ?
+                    ((Project) bindings.get("project")).getMetadata().getName() : args[1];
```

Comment:
Got it!

Reply:
> Project's can actually have the same name. I have 25 projects with unique Project IDs but they all have a common name "clipboard". And that's fine, we allow that. `cross()` however, without a Project name now supplied here will get confused as shown, so we need to improve this handling (in other words, it should know that I am about to perform a `cross()` on the current `binding` to a Project id.

@thadguidry: I know we are not keen on overloading function parameters, but, in this case, I really think we should accept number to identify project. Because, it's just 2 ways of identifying the exact same thing.

What do you think about that?

Regards, A.

Reply:
@antoine2711 Users can be explicit about the project they want to cross with by specifying name.  If the project names are the same, it is trivial for them to rename a project and then be explicit with the names.  So I'd say let's not introduce extra worry about project ids in `cross()` for now.